homeassistant/components/habitica/coordinator.py

```diff
@@ -50,25 +57,27 @@ def __init__(self, hass: HomeAssistant, habitipy: HabitipyAsync) -> None:
                 immediate=False,
             ),
         )
-        self.api = habitipy
-        self.content: dict[str, Any] = {}
+        self.habitica = habitica
+        self.content: ContentData
 
     async def _async_update_data(self) -> HabiticaData:
         try:
-            user_response = await self.api.user.get()
-            tasks_response = await self.api.tasks.user.get()
-            tasks_response.extend(await self.api.tasks.user.get(type="completedTodos"))
-            if not self.content:
-                self.content = await self.api.content.get(
-                    language=user_response["preferences"]["language"]
-                )
-        except ClientResponseError as error:
-            if error.status == HTTPStatus.TOO_MANY_REQUESTS:
-                _LOGGER.debug("Rate limit exceeded, will try again later")
-                return self.data
-            raise UpdateFailed(f"Unable to connect to Habitica: {error}") from error
-
-        return HabiticaData(user=user_response, tasks=tasks_response)
+            user = (await self.habitica.get_user()).data
+            tasks = (await self.habitica.get_tasks()).data
+            completed_todos = (
+                await self.habitica.get_tasks(TaskFilter.COMPLETED_TODOS)
+            ).data
+            if not getattr(self, "content", None):
+                self.content = (
+                    await self.habitica.get_content(user.preferences.language)
+                ).data
```

Comment:
I need to first fetch user data before i can load content, because I need the user's language, otherwise it defaults to english

Reply:
content is 2.7MB, fetching once during coordinator setup. Moving the initial connectivity check to async_setup allows me to fetch content setup, so that's what i'm going to do